src/lib/coda_base/transaction_logic.ml

```diff
@@ -1616,15 +1636,22 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
     Undo.Coinbase_undo.
       { coinbase= cb
       ; previous_empty_accounts= emptys1 @ emptys2
-      ; receiver_timing= receiver_timing_for_undo }
+      ; receiver_timing= receiver_timing_for_undo
+      ; balances=
+          { User_command_status.Balance_data.fee_payer_balance=
```

Comment:
This might cause some confusion as to which balance corresponds to which fee transfer account. maybe make it explicit? (same for coinbase)

Reply:
How do you suggest we do this? We aggregate all transactions at several points, at which stage they all become a `Transaction.t With_status.t`. My preference is to leave this as is and document the behaviour (with unit tests to enforce that we don't change it arbitrarily).

Reply:
`User_command_status.t` (or `Transaction_status.t`) could be a union of 
`User_command {auxiliary:..; user_command_balances:..} | Fee_transfer of Fee_transfer_balances.t | Coinbase of Coinbase_balances.t` Where each of the balance types in turn has the balances associated with the transactions involved. For example: `Fee_transfer_balances.t` could be `fee_receiver1: Balance.t; fee_receiver2: Balance.t option` 
And we always have `empty` for the auxiliary data for fee transfer and coinbase, so that's not required. Maybe create a todo for this and we can discuss this later?

Reply:
I've done the middle-ground that we discussed: the balances are stored in a different, more readable format for the fee-transfers and coinbases, and they are converted where we want a full status instead of their balances concept.